karate-core/src/main/java/com/intuit/karate/Main.java

```diff
@@ -94,6 +94,9 @@
     @Option(names = {"-o", "--output"}, description = "directory where logs and reports are output (default 'target')")
     String output = FileUtils.getBuildDir();
 
+    @Option(names = {"-r", "--htmlreport"}, description = "output html report enabled (default 'true')")
```

Comment:
prefer `--report` if that sounds okay. here is where potentially can add more flags in future similar to `--format`

I was wondering if a system to prefix `~` to disable works, e.g. (not needed now, just seeing if makes sense to you)

`-r ~html,~json` - which means don't emit html *and* JSON reports which are both by default

Reply:
> btw having to wrap each parameter with quotes seems kind of weird... but that's the behavior supported by the function in Karate that parses CLI arguments. was it intended?

Hi @joelpramos it's weird but parsing options works well with short options (i.e: `-e`) but long options (`--env`) needs quoting, this must be something related to picocli

This works fine, but if using long options version it fails
```java
    @Test
    void testParsingCommandLine7() {
        Main options = IdeMain.parseIdeCommandLine("-H com.intuit.karate.cli.VSCodeHook -t ~@ignore -e local -g W:\\dev\\sandbox\\karate-1.0.0-tests\\src\\test\\resources w:\\dev\\sandbox\\karate-1.0.0-tests\\src\\test\\resources\\todos.feature:27");
        assertEquals(1, options.paths.size());
        assertEquals(1, options.tags.size());
        assertEquals(1, options.hookFactoryClassNames.size());
        assertEquals("local", options.env);
        assertEquals("W:\\dev\\sandbox\\karate-1.0.0-tests\\src\\test\\resources", options.configDir);
    }
```

Reply:
also this looks very elegant: `-r ~html,~json`

Reply:
yeah I like that much more elegant

Reply:
@ptrthomas what if I change to the --format instead and remove the new property I added but make sure that html is enabled by default unless explicitly negated (~html)?

Reply:
@joelpramos yes that is an option. but we need to make it clear because this is what is going on
* cucumber json
* junit xml
* karate html
* karate json

so maybe some smart naming convention ? or keep the "karate" domain separate. thoughts

Reply:
I'd keep karate as the "default" namespace and the others maybe we have the name appended:
json
html
junitxml
cucumberjson

It is indeed the karate framework and the other reports it's to provide capabilities for other frameworks.

Reply:
@joelpramos agreed. I'd try to fit a colon maybe e.g. `cucumber:json` but fine otherwise also

Reply:
Makes sense. I was reviewing the usage of the Karate JSON report and I'm not certain whether it can be disabled - the Results object is always instantiated out of the File object where that JSON is stored. I assume it's related with the distributed testing.

We could try to refactor to allow it to be disabled but produce an error if you try using through JobManager with it disabled but I don't really have that set up yet.

It doesn't truly impact my use case as I was just trying not to store anything during a debug session but it's not breaking anything. Maybe I can add a Hook that is executed after the Suite in a debug session to clear any files that are produced?

Reply:
@joelpramos great point thanks. I propose that if the user disables the karate json it will disable the karate html also or at least the summary HTML and printing the final results tally to the sys.out

I was contemplating an option where the JSON would be kept in memory until the end of the test, but not worth it maybe

Reply:
just to clarify - the reason to persist the JSON is not just for distributed testing, but to save memory for a large suite. otherwise we would be accumulating lots of objects just for the sake of the summary report. come to think of it there is a `feature-summary` json you can see used in the `Results` class. maybe that could suffice. agh so many decisions >_<

Reply:
I'll leave it for now and maybe try to create a Hook for the debugger later to clear the created files after the tests are done.  Can probably be enabled by default to the debugger but might even be useful for CI/CD pipelines if we only care about the logs of the execution or if you integrate with an external system (I use report portal and I also zip the HTML report and ship it there).